CI: Failing RestIT on master (nodes.stats/30_discovery/Discovery stats)

This Rest test started failing a few hours ago, I have seen it fail several times on master branch:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+g1gc/415/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+periodic/844/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-intake/583/console

```
:distribution:integ-test-zip:integTest
Executing task ':distribution:integ-test-zip:integTest' (up-to-date check took 0.001 secs) due to:
  Task.upToDateWhen is false.
[ant:junit4] <JUnit4> says kaixo! Master seed: D34B10548736FEA
==> Test Info: seed=D34B10548736FEA; jvm=1; suite=1
Started J0 PID(342@slave-886fdd4f.build.us-west-2a.elasticnet.co).
Suite: org.elasticsearch.test.rest.RestIT
IGNOR/A 0.01s | RestIT.test {p0=update/85_fields_meta/Metadata Fields}
   > Assumption #1: [update/85_fields_meta/Metadata Fields] skipped, reason: [Update doesn't return metadata fields, waiting for #3259] 
  2> REPRODUCE WITH: gradle :distribution:integ-test-zip:integTest -Dtests.seed=D34B10548736FEA -Dtests.class=org.elasticsearch.test.rest.RestIT -Dtests.method="test {p0=nodes.stats/30_discovery/Discovery stats}" -Des.logger.level=WARN -Dtests.security.manager=true -Dtests.jvm.argline="-XX:-UseParNewGC -XX:-UseConcMarkSweepGC -XX:+UseG1GC" -Dtests.locale=el-CY -Dtests.timezone=America/Chihuahua
FAILURE 0.01s | RestIT.test {p0=nodes.stats/30_discovery/Discovery stats} <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: field [nodes] doesn't have a true value
   > Expected: not null
   >      but: was null
   > 	at __randomizedtesting.SeedInfo.seed([D34B10548736FEA:85608EDFE68F0212]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.test.rest.section.IsTrueAssertion.doAssert(IsTrueAssertion.java:48)
   > 	at org.elasticsearch.test.rest.section.Assertion.execute(Assertion.java:69)
   > 	at org.elasticsearch.test.rest.ESRestTestCase.test(ESRestTestCase.java:399)
   > 	at java.lang.Thread.run(Thread.java:745)
  1> [2016-05-18 04:56:46,232][ERROR][org.elasticsearch.test.rest.client] Adding header Accept
  1>  with value application/yaml
IGNOR/A 0.00s | RestIT.test {p0=indices.put_settings/all_path_options/put settings in list of indices}
   > Assumption #1: [indices.put_settings/all_path_options/put settings in list of indices] skipped, reason: [list of indices not implemented yet] 
  1> [2016-05-18 04:57:21,422][ERROR][org.elasticsearch.test.rest.client] Adding header Accept
  1>  with value application/yaml
  2> NOTE: leaving temporary files on disk at: /var/lib/jenkins/workspace/elastic+elasticsearch+master+g1gc/distribution/integ-test-zip/build/testrun/integTest/J0/temp/org.elasticsearch.test.rest.RestIT_D34B10548736FEA-001
  2> NOTE: test params are: codec=Asserting(Lucene60), sim=RandomSimilarity(queryNorm=true,coord=no): {}, locale=el-CY, timezone=America/Chihuahua
  2> NOTE: Linux 2.6.32-573.el6.x86_64 amd64/Oracle Corporation 1.8.0_71 (64-bit)/cpus=4,threads=1,free=275366168,total=536870912
  2> NOTE: All tests run in this JVM: [RestIT]
Completed [1/1] in 73.34s, 519 tests, 1 failure, 2 skipped <<< FAILURES!

Tests with failures:
  - org.elasticsearch.test.rest.RestIT.test {p0=nodes.stats/30_discovery/Discovery stats}
```

Also reproduces locally when I run `gradle clean check` in `distribution/integ-test-zip`, but cannot reproduce with the above "REPRODUCE WITH" line.

--- a/core/src/main/java/org/elasticsearch/monitor/fs/FsInfo.java
+++ b/core/src/main/java/org/elasticsearch/monitor/fs/FsInfo.java
@@ -396,12 +396,14 @@ public class FsInfo implements Iterable<FsInfo.Path>, Writeable, ToXContent {
                     builder.endObject();
                 }
                 builder.endArray();
+
                 builder.startObject("total");
                 builder.field(OPERATIONS, totalOperations);
                 builder.field(READ_OPERATIONS, totalReadOperations);
                 builder.field(WRITE_OPERATIONS, totalWriteOperations);
                 builder.field(READ_KILOBYTES, totalReadKilobytes);
                 builder.field(WRITE_KILOBYTES, totalWriteKilobytes);
+                builder.endObject();
             }
             return builder;
         }

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/nodes.stats/30_discovery.yaml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/nodes.stats/30_discovery.yaml
@@ -1,9 +1,5 @@
 ---
 "Discovery stats":
-  - skip:
-      version:     "5.0.0 - "
-      reason:      Tracked in issue 18433 
-
   - do:
       cluster.state: {}
 

